tiledb/sm/cpp_api/deleter.h

```diff
@@ -118,11 +118,11 @@ class Deleter {
   }
 
   void operator()(tiledb_current_domain_t* p) const {
-    tiledb_current_domain_free(&p);
+    tiledb_current_domain_free(ctx_->ptr().get(), &p);
```

Comment:
The lifetime management complications I am referring to are that we need to ensure that the context must outlive the current domain and the other objects that require it to be freed.

I had managed to do that in C# (which was not without performance regressions, because freeing all objects would require multiple garbage collections unless the user explicitly disposed them), but I am not seeing it here. `Deleter::ctx_` is a raw pointer, and `CurrentDomain::ctx_` is a reference wrapper.